components/script/dom/writablestreamdefaultcontroller.rs

```diff
@@ -268,37 +268,58 @@ impl Callback for WriteAlgorithmRejectionHandler {
 
 /// The type of sink algorithms we are using.
 #[derive(JSTraceable, PartialEq)]
+#[cfg_attr(crown, crown::unrooted_must_root_lint::must_root)]
 pub enum UnderlyingSinkType {
     /// Algorithms are provided by Js callbacks.
-    Js,
+    Js {
+        /// <https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-abortalgorithm>
+        abort: RefCell<Option<Rc<UnderlyingSinkAbortCallback>>>,
+
+        start: RefCell<Option<Rc<UnderlyingSinkStartCallback>>>,
+
+        /// <https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-closealgorithm>
+        close: RefCell<Option<Rc<UnderlyingSinkCloseCallback>>>,
+
+        /// <https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-writealgorithm>
+        write: RefCell<Option<Rc<UnderlyingSinkWriteCallback>>>,
+    },
     /// Algorithms supporting streams transfer are implemented in Rust.
     /// The promise and port used in those algorithms are stored here.
     Transfer {
         backpressure_promise: Rc<RefCell<Option<Rc<Promise>>>>,
         port: Dom<MessagePort>,
     },
+    /// Algorithms supporting transform streams are implemented in Rust.
+    #[allow(unused)]
+    Transform(/*Dom<TransformStream>, Rc<Promise>*/),
```

Comment:
Surprised that this compiles, I'd though you'd have to comment out the braces as well...